File: app/src/main/kotlin/net/tigr/soulcalc/SoulCalcApp.kt
/*
 * SPDX-License-Identifier: MIT
 * Copyright (c) 2026 Albert Zenkoff
 */

package net.tigr.soulcalc

import android.app.Application
import net.tigr.soulcalc.data.db.AppDatabase
import net.tigr.soulcalc.data.preferences.ThemePreferences
import net.tigr.soulcalc.data.repository.SheetRepository

class SoulCalcApp : Application() {

    val database: AppDatabase by lazy {
        AppDatabase.getInstance(this)
    }

    val sheetRepository: SheetRepository by lazy {
        SheetRepository(
            sheetDao = database.sheetDao(),
            lineDao = database.lineDao()
        )
    }

    val themePreferences: ThemePreferences by lazy {
        ThemePreferences(this)
    }
}

File: ios/SoulCalc/SoulCalcApp.swift
// SPDX-License-Identifier: MIT
// Copyright (c) 2026 Albert Zenkoff

import SwiftUI

@main
struct SoulCalcApp: App {
    @StateObject private var themePreferences = ThemePreferences()
    @StateObject private var viewModel = MainViewModel()

    var body: some Scene {
        WindowGroup {
            MainScreen(
                viewModel: viewModel,
                themeMode: $themePreferences.themeMode
            )
            .preferredColorScheme(themePreferences.themeMode.colorScheme)
        }
    }
}
